.gitignore

```diff
@@ -1,3 +1,12 @@
+hypothesis_logs/
+*.log
+logs/
+__pycache__/
+generated_hypotheses/
+function_calling_results/
+hypothesis_results/*
+AG4/*
```

Comment:
I think we need to add `**/version.py` for `setuptools-scm`'s made `version.py` file